### PR TITLE
add simple implement of WebView for win32

### PR DESCRIPTION
**Pull request by summerinsects**
Is it being implemented by official developing team? If yes, ignore it

This implement contains some simple functions such as loadUrl, loadFile,
evaluateJS
Other more functions have not been implemented yet
My environment is win8.1 vs2013, and it works well

### CODE REVIEW

**Comment by minggo** [comment]
Thanks.
No, we are not doing right now.

Will you continue to implement other functions?

**Comment by summerinsects** [comment]
@minggo I will try my best.

**Comment by minggo** [comment]
Thanks.

**Comment by minggo** [comment]
You should exclude `UIWebViewImpl_win32.cpp` to be compiled on linux.
You can do it in two ways:
- modify `cocos/CMakeLists.txt` to exclude `UIWebViewImpl_win32.cpp`
- make `UIWebViewImpl_win32.cpp` can be included in all platforms, just add macros at begin and end like this

``` c++
// UIWebViewImpl_win32.cpp
#if CC_TARGET_PLATFROM == CC_PLATFORM_WINDOWS
...
#endif
```

**Comment by summerinsects** [comment]
OK

File: cocos/ui/UIWebViewImpl_win32.cpp [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+    class Win32WebView
+    {
+    public:
+        Win32WebView()
+            : m_pWebBrowser2(NULL)
+            , m_bCanGoBack(true)
+            , m_bCanGoForward(true)
+            , m_dwCookie(0)
+            , m_ulRef(0)
```
Comment (minggo) [inline cocos/ui/UIWebViewImpl_win32.cpp]: Please use _xxx as member variables to obey cocos2d-x programming guide.

File: cocos/ui/UIWebViewImpl_win32.cpp [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        {
+            lazyInit();
+        }
+
+        bool createWebView()
+        {
+            bool bRet = false;
```
Comment (minggo) [inline cocos/ui/UIWebViewImpl_win32.cpp]: Please don't use Hungary style.
bRet -> ret

**Comment by summerinsects** [comment]
These codes is form an old win32 project.
OK, I'll modify them later

**Comment by minggo** [comment]
Thanks.

**Comment by summerinsects** [comment]
@minggo 
Cound I pass nullptr to C-style win32 api functions, such as GetModuleHandle,
or initialize some handle with nullptr, such as HWND,
or compare handle with nullptr to check whether some function call has no error?

**Comment by minggo** [comment]
I am not sure. I haven't noticed that it is win32 API, so use NULL is ok.

**Comment by summerinsects** [comment]
Then, I keep using NULL when calling win32 APIs or doing some operation with handle?

**Comment by minggo** [comment]
Yep, using NULL for win32 API.

**Comment by summerinsects** [comment]
I have added more implementation.
Since I have no IOS environment, I built an apk, and studied the 3 callbacks.
However, due to platform differences, there should be many differences when callback is called

**Comment by minggo** [comment]
> Since I have no IOS environment, I built an apk, and studied the 3 callbacks.
> However, due to platform differences, there should be many differences when callback is called

You are implementing for windows, is there any problem related to iOS?

**Comment by summerinsects** [comment]
I want to make the platform differences as small as possible

**Comment by minggo** [comment]
Got it.
Is this PR ready to be merged?

**Comment by summerinsects** [comment]
Yes

**Comment by summerinsects** [comment]
@minggo 
It seems that my version is too old. Should I recheckout and PR again?

**Comment by minggo** [comment]
You are appreciated if you recheckout and PR again.
And please add files into VS project and make the test case work.

**Pull request closed without merging**